src/Rules/NoFirstOnSingularModelRule.php

```diff
@@ -0,0 +1,95 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Larastan\Larastan\Rules;
+
+use Illuminate\Database\Eloquent\Model;
+use PhpParser\Node;
+use PhpParser\Node\Expr\MethodCall;
+use PhpParser\Node\Identifier;
+use PHPStan\Analyser\Scope;
+use PHPStan\Rules\Rule;
+use PHPStan\Rules\RuleError;
+use PHPStan\Rules\RuleErrorBuilder;
+use PHPStan\ShouldNotHappenException;
+use PHPStan\Type\ObjectType;
+use PHPStan\Type\Type;
+use PHPStan\Type\UnionType;
+
+/** @implements Rule<MethodCall> */
+final class NoFirstOnSingularModelRule implements Rule
+{
+    /**
+     * Specifies the node type this rule is interested in.
+     * We are interested in method calls.
+     *
+     * @return class-string<Node>
+     */
```

Comment:
There is no need to duplicate the lengthy phpdocs---you can actually remove them completely, but if you insist on having them then just use:


```suggestion
    /** @inheritDoc */
```